components/script/dom/htmlscriptelement.rs

```diff
@@ -1054,19 +1063,18 @@ impl HTMLScriptElement {
                 } else {
                     document.set_current_script(Some(self))
                 }
-            },
-            ScriptType::Module => document.set_current_script(None),
-        }
-
-        match script.type_ {
-            ScriptType::Classic => {
                 self.run_a_classic_script(&script, can_gc);
                 document.set_current_script(old_script.as_deref());
             },
             ScriptType::Module => {
+                document.set_current_script(None);
                 assert!(document.GetCurrentScript().is_none());
```

Comment:
Why is this change necessary? The assertion right below is now redundant, so one of these lines should be removed.redundant

Reply:
I merge two `match`es because they are all in the same step 6. I'll remove the assert since it's redundant indeed.